pilot/pkg/security/model/authentication.go

```diff
@@ -157,17 +159,23 @@ func ApplyToCommonTLSContext(tlsContext *tls.CommonTlsContext, proxy *model.Prox
 				},
 			}
 		}
+		sdsSecretConfig := ConstructSdsSecretConfig(model.GetOrDefault(res.GetRootResourceName(), SDSRootResourceName))
+		if proxy.Type == model.Router {
```

Comment:
Do we want the feature flag to apply to sidecars that use file based secrets? If so, then maybe this conditional needs to be relaxed? If we do want it to only apply to gateways, then we should probably rename the flag to indicate that